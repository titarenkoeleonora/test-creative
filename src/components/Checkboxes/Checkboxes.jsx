import {useState} from "react";

import animals from "../../mocks/mocks";

const Checkboxes = () => {
  const [checkedInputs, setCheckedInputs] = useState([false, false, false]);
  const [checkboxAllChecked, setCheckboxAllChecked] = useState(false);

  const getCheckedInputs = (evt, index) => {
    const value = evt.target.checked;

    setCheckedInputs([...checkedInputs.slice(0, index), value, ...checkedInputs.slice(index + 1)]);
    setCheckboxAllChecked(false);
  };

  const checkAllCheckbox = () => {
    setCheckboxAllChecked(!checkboxAllChecked);

    if (checkboxAllChecked === false) {
      setCheckedInputs([true, true, true]);
    } else {
      setCheckedInputs([false, false, false]);
    }
  };

  return (
    <section className="checkboxes">
      <div className="container">
        <form className="checkboxes-form checkboxes__form" method="get" action="#">
          <p>Ваше любимое животное:</p>
          <ul className="checkboxes-form__list">
            <li className="checkboxes-form__item">
              <input
                className="checkbox checkboxes-form__checkbox visually-hidden"
                type="checkbox"
                id="all"
                name="all"
                onChange={checkAllCheckbox}
                checked={checkboxAllChecked}
              />
              <label className="checkboxes-form__label" htmlFor="all" >Все</label>
            </li>

            {animals.map((animal, index) => (
              <li className="checkboxes-form__item" key={animal + index}>
                <input
                  className="checkbox checkboxes-form__checkbox visually-hidden"
                  type="checkbox"
                  id={animal.id + index}
                  name={animal.id}
                  onChange={(evt) => getCheckedInputs(evt, index)}
                  checked={checkedInputs[index]}
                />
                <label className="checkboxes-form__label" htmlFor={animal.id + index} >{animal.title}</label>
                {checkedInputs[index] === true ?
                  <img className="checkboxes-form__animal" src={`./images/${animal.id}.png`} width="50" height="50" alt={animal.title}/>
                  : ""
                }
              </li>
            ))}
          </ul>
        </form>
      </div>
    </section>
  );
};

export default Checkboxes;