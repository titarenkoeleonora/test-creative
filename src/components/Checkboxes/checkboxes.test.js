import React from "react";
import renderer from "react-test-renderer";
import Checkboxes from "./Checkboxes";

const animals = [
  {
    id: "dog",
    title: "Собака",
  },
  {
    id: "cat",
    title: "Кот",
  },
  {
    id: "mouse",
    title: "Мышь",
  },
];

it('Render Checkboxes component', () => {
  const tree = renderer
    .create(
      <Checkboxes
        animals={animals}
      />
    );

  expect(tree).toMatchSnapshot();
});