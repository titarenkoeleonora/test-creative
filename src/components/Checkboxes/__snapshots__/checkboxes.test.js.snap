// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Checkboxes component 1`] = `
<section
  className="checkboxes"
>
  <div
    className="container"
  >
    <div
      className="checkboxes__wrapper"
    >
      <h2
        className="checkboxes__title"
      >
        Отображение/скрытие элементов при переключении чек-боксов
      </h2>
      <form
        action="#"
        className="checkboxes-form checkboxes__form"
        method="get"
      >
        <p
          className="checkboxes-form__title"
        >
          Ваше любимое животное:
        </p>
        <ul
          className="checkboxes-form__list"
        >
          <li
            className="checkboxes-form__item"
          >
            <input
              checked={false}
              className="checkbox checkboxes-form__checkbox visually-hidden"
              id="all"
              name="all"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="checkboxes-form__label"
              htmlFor="all"
            >
              Все
            </label>
          </li>
          <li
            className="checkboxes-form__item"
          >
            <input
              checked={false}
              className="checkbox checkboxes-form__checkbox visually-hidden"
              id="dog0"
              name="dog"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="checkboxes-form__label"
              htmlFor="dog0"
            >
              Собака
            </label>
            
          </li>
          <li
            className="checkboxes-form__item"
          >
            <input
              checked={false}
              className="checkbox checkboxes-form__checkbox visually-hidden"
              id="cat1"
              name="cat"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="checkboxes-form__label"
              htmlFor="cat1"
            >
              Кот
            </label>
            
          </li>
          <li
            className="checkboxes-form__item"
          >
            <input
              checked={false}
              className="checkbox checkboxes-form__checkbox visually-hidden"
              id="mouse2"
              name="mouse"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="checkboxes-form__label"
              htmlFor="mouse2"
            >
              Мышь
            </label>
            
          </li>
        </ul>
      </form>
    </div>
  </div>
</section>
`;
